src/ansys/geometry/core/designer/component.py

```diff
@@ -174,15 +175,24 @@ def __init__(
         if preexisting_id:
             self._name = name
             self._id = preexisting_id
+            self._intance_name = instance_name
         else:
+            print("no preexisting_id")
             if parent_component:
                 template_id = template.id if template else ""
                 new_component = self._component_stub.Create(
-                    CreateRequest(name=name, parent=parent_component.id, template=template_id)
+                    CreateRequest(
+                        name=name,
+                        parent=parent_component.id,
+                        template=template_id,
+                        instance_name=instance_name,
+                    )
                 )
+
                 # Remove this method call once we know Service sends correct ObjectPath id
                 self._id = new_component.component.id
                 self._name = new_component.component.name
+                self._intance_name = instance_name
```

Comment:
```suggestion
                self._instance_name = new_component.component.instance_name
```
You should get this value from the response of the creation.

Reply:
I did it this way for a reason. In API server we create the component but then return an occurence of which doesn't share the instance name. @b-matteo

Reply:
I still think we should use the response value. The occurrence returned should contain that instance name IMO.

Reply:
I think we can merge it this way because it is time sensitive and then we should have a talk with Matteo to make sure it is not against the intended occurrence design there.

Reply:
I get it is time sensitive but we should do things properly and not rush things. Otherwise we end up like in #1396 - which we haven't solved in weeks already.

Reply:
@b-matteo, @RobPasMue. I'm done with this PR, I'll let you guys decide what you want to do. Feel free to update it. Thanks.

Reply:
@umutsoysalansys - a PR is not done until it is merged.